packages/vite/src/node/plugins/terser.ts

```diff
@@ -90,19 +90,19 @@ export function terserPlugin(config: ResolvedConfig): Plugin {
         return null
       }
 
-      // Do not minify ES lib output since that would remove pure annotations
-      // and break tree-shaking.
-      if (config.build.lib && outputOptions.format === 'es') {
-        return null
-      }
-
       // Lazy load worker.
       worker ||= makeWorker()
 
       const terserPath = loadTerserPath(config.root)
       const res = await worker.run(terserPath, code, {
         safari10: true,
         ...terserOptions,
+        format: {
+          ...terserOptions.format,
+          // For ES lib mode, preserve comments to keep pure annotations for tree-shaking
+          preserve_annotations:
+            config.build.lib && outputOptions.format === 'es',
+        },
```

Comment:
I think we should also respect the user's `preserve_annotations` config if they're not building a library in ES output. E.g. if they enable `preserve_annotations` explicitly for other things, this would've always force it to `false`.

Reply:
Ah, good catch! I've fixed that in 51ad1841d6d129f659be04fa2724c212b0062850